implementing-nosto/implement-search/implement-search-using-code-editor/implementing-autocomplete.md

```diff
@@ -44,10 +44,13 @@ The full list of Configuration options is documented [here](https://nosto.github
 
 {% code title="autocomplete/index.jsx" %}
 ```jsx
-import { useAppState, AutocompleteElement } from '@nosto/preact'
+import { useAppStateSelector, AutocompleteElement } from '@nosto/preact'
 
 export default () => {
-    const { response: { products, keywords } } = useAppState()
+    const { products, keywords } = useAppStateSelector((state) => ({
+        products: state.response.products,
+        keywords: state.response.keywords
+    }))
```

Comment:
Note: Could use the Pick utility here, but I think simplicity is better in the docs.